src/useDraggable.ts

```diff
@@ -138,24 +139,29 @@ export function useDraggable<T>(...args: any[]): UseDraggableReturn {
    * @param {DraggableEvent} evt
    */
   function onAdd(evt: DraggableEvent) {
-    const element = evt.item[CLONE_ELEMENT_KEY]
-    if (isUndefined(element)) return
-    removeNode(evt.item)
-    insertElement(unref(list), evt.newDraggableIndex!, element)
+    const fromList = getListFromEl(evt.from);
+    // get the object from the original list to get the same object instance, to preserve reactivity
+    const item = fromList.value[evt.oldIndex!];
+    if (isUndefined(item)) return
+    const newList = [...list.value];
+    insertElement(newList, evt.newDraggableIndex!, item);
+    list.value = newList;
+  }
+
+  function getListFromEl(el: HTMLElement) {
+    const key = Object.keys(el).find(key => key.startsWith('Sortable'));
+    return key && (el as any)[key]?.__list;
```

Comment:
This is a little bit ugly, but I need to have access to the original list to get the exact same object, not a clone. Cloning removes any getter and setter logic, thus kills reactivity.

Reply:
But at least with this approach we rely on native Vue rerender instead of DOM manipulation?